### PR TITLE
Implement advanced local movement

### PR DESCRIPTION
**Pull request by minecrawler**
Cleaned #342 , implements movement helpers on `LocalTransform`

### CODE REVIEW

File: amethyst_renderer/src/scene.rs [hunk 1]
```diff
@@ -65,8 +65,8 @@ impl Scene {
     /// Returns the active camera in the scene.
     ///
     /// TODO: Render to multiple viewports with possibly different cameras.
-    pub fn active_camera(&self) -> Option<&Camera> {
-        self.cameras.first()
+    pub fn active_camera(&mut self) -> Option<&mut Camera> {
```
Comment (Rhuagh) [inline amethyst_renderer/src/scene.rs]: Why is this must?

Comment (Rhuagh) [reply amethyst_renderer/src/scene.rs]: Mut

Comment (minecrawler) [reply amethyst_renderer/src/scene.rs]: I guess it's not. It was, when I put all the movement on `Camera` in the beginning. Should I remove it?

Comment (Rhuagh) [reply amethyst_renderer/src/scene.rs]: Are we even using scene?  I think it might be dead code tbh, so ignore it

Comment (Rhuagh) [reply amethyst_renderer/src/scene.rs]: Umm, scene.rs don't exist in the develop branch

Comment (Rhuagh) [reply amethyst_renderer/src/scene.rs]: No sorry, the file exist but it's not used. So ignore it.

**Comment by Rhuagh** [comment]
I created an issue for the dead code

File: amethyst_core/src/transform/components/local_transform.rs [hunk 2]
```diff
@@ -1,22 +1,31 @@
...
     pub rotation: [f32; 4],
     /// Scale vector [x, y, z]
     pub scale: [f32; 3],
+    /// Translation/position vector [x, y, z]
+    pub translation: [f32; 3],
 }
 
 impl LocalTransform {
+    /// Rotate to look at a point in space (without rolling)
+    pub fn look_at(&mut self, orientation: &Orientation, position: Vector3<f32>) -> &mut Self {
```
Comment (torkleyy) [inline amethyst_core/src/transform/components/local_transform.rs]: I'd like to note that with this PR we would expose cgmath types.

Comment (torkleyy) [reply amethyst_core/src/transform/components/local_transform.rs]: I'm not sure if that's a good thing.

File: amethyst_core/src/transform/components/local_transform.rs [hunk 3]
```diff
@@ -33,6 +42,122 @@ impl LocalTransform {
...
+        let rel_axis_normalized = Quaternion::from(self.rotation)
+            .rotate_vector(Vector3::from(axis))
+            .normalize();
+        let q = Quaternion::from_axis_angle(rel_axis_normalized, angle);
+
+        self.rotate(q)
+    }
+
+    /// Set the position.
+    pub fn set_position(&mut self, position: Point3<f32>) -> &mut Self {
```
Comment (torkleyy) [inline amethyst_core/src/transform/components/local_transform.rs]: E.g. here I have the feeling that exposing the cgmath type just makes it more verbose.

File: amethyst_core/src/transform/components/local_transform.rs (hunk 2 above)
Comment (minecrawler) [reply amethyst_core/src/transform/components/local_transform.rs]: We already expose cgmath in other places as well (see the camera object as an example). However, I also think that mint would be a better idea. I've been thinking about [streamlining the API with traits and Mint](https://github.com/minecrawler/amethyst/blob/feature/enhanceTransformHelpers/amethyst_core/src/movement.rs) and expanding the functionality to global transformation. The question for me would be: Should I put in Mint types here, or do so in the other PR with the traits (if they are acceptable)?

Comment (torkleyy) [reply amethyst_core/src/transform/components/local_transform.rs]: I don't really mind as long as it gets done before the release.

**Comment by Rhuagh** [comment]
I opened an issue for removing cgmath from the public facing APIs (#452). Can we approve this then?

**Comment by Rhuagh** [comment]
Failed internally in appveyor on rustc download. 

bors r+

**Pull request merged**